### DISCUSSION TITLE
Endless Bootloop

### DISCUSSION THREAD

**Discussion by indifferent1** [gtxaspec/wz_mini_hacks#166] [Q&A]
First off, THANK YOU, if I can get this working, its exactly what I needed.

I have gone through the installation process several times on a v2. Ive gone as far as supplying the firmware directly into the compile_image script to make this work. I have also reverted to factory firmware and successfully booted, and initialized the camera. However, every time I try to use the modified firmware, it fails to boot, ending up in an endless boot loop. Does anyone have any suggestions?

**Comment by gtxaspec** [comment]
@indifferent1 in wz_mini/logs are there any logs created?

**Reply by indifferent1** [reply]
It does not appear to have made it that far. No logs in the log directory.

**Reply by gtxaspec** [reply]
when you create the demo.bin, using the default method which downloads the firmware, how big is the resulting demo.bin? in bytes please, also, are you changing any config variables in wz_mini.conf?

**Reply by indifferent1** [reply]
Standy, I am going through the process on another linux box. I modified the wz_mini.conf once, but still have the same issue.

**Reply by indifferent1** [reply]
> when you create the demo.bin, using the default method which downloads the firmware, how big is the resulting demo.bin? in bytes please, also, are you changing any config variables in wz_mini.conf?

Without providing the firmware I downloaded from Wyze, the compile_image.sh fails without warning, here's the log.
[log.txt](https://github.com/gtxaspec/wz_mini_hacks/files/9069974/log.txt)

Creating the demo.bin with the firmware provided is 10.6M

**Reply by gtxaspec** [reply]
gotcha, and when you provide the v2 1002 zip, named demo.zip ?

**Reply by indifferent1** [reply]
here is the log, when the file is provided.
[log2.txt](https://github.com/gtxaspec/wz_mini_hacks/files/9070165/log2.txt)

**Reply by indifferent1** [reply]
Tried it again a few times with the same results. Not sure what I am doing wrong.

**Reply by clee604** [reply]
What sd card are you using?

**Reply by indifferent1** [reply]
> What sd card are you using?

I have tried several

**Reply by clee604** [reply]
Just checking.. but they've been formatted to FAT32 and not ex-FAT?

**Reply by indifferent1** [reply]
> Just checking.. but they've been formatted to FAT32 and not ex-FAT?

Yeah, fat32, otherwise I would not have been able to install the factory firmware either.

**Reply by gtxaspec** [reply]
just tested the latest release on a v2, working fine on two V2's i just flashed.  Any lights flashing on the camera? What firmware were you running before you flashed? try this

**Reply by clee604** [reply]
I had the bootloop a few times when I trying to get ethernet to work on my V2, but it usually works fine with wifi.. 

from what I figured out, the camera reboots itself when it can't get a connection for whatever reason which leads to this bootloop. But in my situation I get the wz_mini first boot audio prompts and the log files get generated before it reboots which doesn't seem to be the case for you.

Not sure if you have a unifi AP, but I vaguely remembering having to disable PMF on my wifi network but I don't remember if it was for this firmware or for something else.

**Reply by gtxaspec** [reply]
@clee604 thanks for reminding me, I will try to enable PMF on the wifi drivers, I had this on my list before but I had forgotten.

**Reply by indifferent1** [reply]
I do have unifi, @gtxaspec is that the modified firmware? The only light I get, is the orange one and it blinks one time when it reboots

**Reply by gtxaspec** [reply]
yes, its the output from compile_image.sh

**Reply by indifferent1** [reply]
> yes, its the output from compile_image.sh

Well, I loaded that into the sdscard, its still looping. I will go thru the whole process again, starting with fresh firmware and I will try using your output. @clee604 I checked PMF on my unifi controller, and it is disabled.

**Reply by gtxaspec** [reply]
confirming, what color v2, and what are the first three groups of the Mac address?

**Reply by indifferent1** [reply]
@gtxaspec  Its black, first three groups are 2C:AA:8E

**Reply by gtxaspec** [reply]
After you flash the modified firmware, and it just stalls, does anything happen after a while, if you press the setup button on the camera?

**Reply by indifferent1** [reply]
@gtxaspec I left it for an hour last night to see if anything happened on its own after flashing. It does not change. Pressing the button yields no result, and holding the button through a reboot causes it to reload the firmware again.

**Reply by gtxaspec** [reply]
do you have another camera to test? have you tried different SD cards it you have them to test?  quite the unusual situation.

**Reply by indifferent1** [reply]
I have another black one I purchased at the same time. I will try it on that. I have tried different SD cards as well.

**Reply by indifferent1** [reply]
@gtxaspec same results on the second camera. I dont know if it makes a difference but I am working with Mac. I will try another computer later. Thanks

**Reply by indifferent1** [reply]
@gtxaspec Just tried it with a straight up windows computer and it still boot loops.

**Reply by gtxaspec** [reply]
copy.  I am investigating this further, stand by.  I'll try to add some logging early in the boot process to try and find out why this is happening.

**Reply by indifferent1** [reply]
Ok, thank you for working on this. I am standing by, and should be available most of the evening.

**Reply by indifferent1** [reply]
@gtxaspec is there anything I can do to pull a serial console on it via usb?

**Reply by gtxaspec** [reply]
> @gtxaspec is there anything I can do to pull a serial console on it via usb?
@indifferent1 unfortunately not, you would have to open the camera and hook on to the uart serial headers to get some output

**Reply by gtxaspec** [reply]
try this.  flash the demo.bin, and rename the wz_mini folder that's in the root of the sd card to wz_mini_old

does the camera boot?

**Reply by indifferent1** [reply]
No it does not, I also just tried the openmiko firmware, which did successfully boot

**Reply by gtxaspec** [reply]
really strange, something in the kernel it seems.

**Reply by clee604** [reply]
have you tried with an older version of wz_mini?
https://github.com/gtxaspec/wz_mini_hacks/tree/66031b41f53ad5578c31c2b9a56e52211fdac85d

I haven't been able to figure out what it is or what the cause is.. but newer versions (some kernel changes??) seem to take longer to boot and ethernet didn't work with some newer versions on my V2.

**Reply by indifferent1** [reply]
I can give it a try, im playing with the openmiko version and not having much luck with it, I dont think its got the right camera sensor. Is there any information I can pull from here before I try the older version of wz_mini?

**Reply by gtxaspec** [reply]
@indifferent1 if you can get me the `dmesg` log from openmiko, i can try and see if any hardware is different on your camera versions.

**Reply by indifferent1** [reply]
See attached dmesg output
[dmesg.log](https://github.com/gtxaspec/wz_mini_hacks/files/9075891/dmesg.log)

**Reply by gtxaspec** [reply]
everything looks good.  let me implement the logging on the wz_mini side, i'm testing the changes now

**Reply by indifferent1** [reply]
@gtxaspec I am stepping away for a little bit and awaiting your testing.

**Comment by gtxaspec** [comment]
@indifferent1 
[v2_kernel.bin.gz](https://github.com/gtxaspec/wz_mini_hacks/files/9075903/v2_kernel.bin.gz)
extract this gzip, and replace the v2_kernel.bin in the v2_install folder with this one.  then run compile_image.sh and flash the demo.bin.

boot the camera, wait a while, if it still doesn't boot up, after a while, pull the card, the log will be in the root of the sd card, named wz_mini_initramfs.log

**Reply by indifferent1** [reply]
Got it, standby doing it now

**Reply by indifferent1** [reply]
It did not download the firmware again, i have to add the zip file manually

**Reply by indifferent1** [reply]
No log file was generated after 5 cycles. each cycle now concludes with a blue flicker of the light however. and the new demo.bin is also 11M

**Reply by indifferent1** [reply]
and its growing, the log must be in there

**Reply by gtxaspec** [reply]
hmm, no it wouldn't be in the demo.bin.  something else is going on.

**Reply by indifferent1** [reply]
Im not sure what to add here....

**Reply by n5tti** [reply]
Check this thread here -- https://github.com/gtxaspec/wz_mini_hacks/discussions/104
I was able to get mine working by downloading a copy of the v2 firmware, creating a folder called v2_ro and putting the .bin file there.
Once I did that, I ran compile_image.sh again, and got  a working cam!
Thanks

**Reply by AnthonyMaiorani** [reply]
I'm having the same issue, also with a black v2. 

Tried the installation process with Windows, Ubuntu, and macOS. Tried formatting with several different utilities, tried manually providing the bin file both in the v2_install and v2_ro folder. All resulted in infinite boot looping. 

Yellow light goes on then off, IR cut filter switches back and forth with it. Voice prompt never happens. I left it for over an hour and no change. 

As far as I can tell from a quick search through existing issues and discussions, nobody has successfully installed this on a black v2. Figuring it was an incompatibility specifically with the black model, I took the SD card out and stuck it in a white v2 making no changes. It worked first try. 

I'd be happy to help provide any information you need to get this working on the black model.

**Reply by Bob-AOL** [reply]
Any thoughts on the same problem existing with the new installation approach since the demo.bin script isn't in the mix?

Two of my three V2s simply won't even acknowledge the latest wz_mini version..., or something required to recognize it. No logs. Nothing.

The good news is that the new installation approach (Discussion #241) still allows for reverting to other firmware after the fact and I was able to reload earlier OEM firmware and the last version of the Wyze RTSP effort on both problem cams. All flashed quite normally as one would expect. Absolutely nothing from the wz_mini installation attempts on these cameras even before the new scrub and reload approach -- which I was anxiously hoping would be the magic bullet to remedy this particular situation.

FWIW, all three of the cams came online in Dec. 2020 if there's any significance to the date. Maybe it has something to do with the security issues identified by the Bit Defender folks awhile back?

**Comment by Bob-AOL** [comment]
Two of my three recently acquired white V2s exhibit the same behavior. The third works like a charm with either of the same memory cards and existing files from the other cameras. The problem units appear to be newer -- at least in appearance. Just got them (used) and am still trying to figure out what I actually have and what the differences may be.

(Discussion #241 ) May have some potential once folks have a chance to kick the tires and take it for a spin around the block. Beyond my pay grade to analyze it, but it worked like a charm for my one cam.

**Comment by Bob-AOL** [comment]
Has there been anything new on this -- perhaps in another discussion elsewhere? I'm running out of bullets trying to get my two uncooperative white V2s mini-ized with my limited knowledge of what variables may be involved or what else to try.

I did successfully implement the #241 installation on all three V2s with one taking wz_mini flawlessly. The other two work fine with the latest OEM 1002 firmware as well as the last Wyze RTSP offering -- and back again. Just won't boot wz_mini.

**Reply by bjs-pdx** [reply]
Just tried installing on an older V2 (2019 vintage) and installation failed.  Also unable to flash back to OEM firmware.  Get a yellow light then nothing.  Did you have any issues flashing back to OEM?

**Reply by Bob-AOL** [reply]
Nope. Was even able to flash the last version of the Wyze RTSP firmware without issue. Normal behavior re-flashing on cams that used both the old and new method of installing wz_mini -- which also wouldn't install in either case. Pulled the SD card. Replaced the wz_mini files with an appropriate demo.bin file and did the normal flash routine holding down the reset button while applying power. Cycled from yellow to the blue sequence like normal until booted and operational. No hic-cups.

**Reply by bjs-pdx** [reply]
Never mind...my stupidity.  SD card wasn't FAT32.  Once I put in the correctly formatted card I was able to go back to the RTSP version from Wyze.  However, the wz_mini never loaded with the correctly formatted disk.  So going to submit an issue on that.
